### PR TITLE
[Distributed] use implicit to detect compiler made fields

### PR DESCRIPTION
**Pull request by ktoso**
Work on rdar://95658795

The detection of "please don't declare those explicitly" is broken cross module, because the `synthesized` flag is never serialized or carried into swift interfaces. Instead we can rely on other flags.

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please build toolchain

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 1]
```diff
@@ -79,8 +81,8 @@ static VarDecl*
 //
 // The "derived" mechanisms are not really geared towards emitting for
 // what already has a witness.
-static VarDecl *addImplicitDistributedActorIDProperty(
-    ClassDecl *nominal) {
+static VarDecl *addImplicitDistributedActorIDProperty(ClassDecl *nominal) {
+  fprintf(stderr, "[%s:%d] (%s) add property to %s \n", __FILE__, __LINE__, __FUNCTION__, nominal->getNameStr().str().c_str());
```
Comment (DougGregor) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:85]: Stray `fprintf`.

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 2]
```diff
@@ -37,40 +37,42 @@ using namespace swift;
...
+                                               DeclName name) {
+  assert(decl && "decl was null");
+  auto &C = decl->getASTContext();
+
+  auto clazz = dyn_cast<ClassDecl>(decl);
+  if (!clazz)
+    return nullptr;
+
+  auto refs = decl->lookupDirect(name);
+  if (refs.size() != 1)
```
Comment (DougGregor) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:50]: This is going to break if, say, some extension somewhere adds an unrelated function with the same name. Usually we walk the lookup results to find what we're looking for, ignoring anything with the same name.

It doesn't have to be addressed right now, but perhaps in the future.

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 3]
```diff
@@ -37,40 +37,42 @@ using namespace swift;
...
+    expectedType = getDistributedActorSystemType(decl);
+  } else {
+    llvm_unreachable("Unexpected distributed actor property lookup!");
+  }
+
+  if (!var->getInterfaceType()->isEqual(expectedType))
+    return nullptr;
+
+  if (!var->isSynthesized()) {
+    var->dump();
```
Comment (DougGregor) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:70]: I don't think we want the "dump" here. Also, did you mean to check `isImplicit` like the PR text implies?

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 4]
```diff
@@ -37,40 +37,42 @@ using namespace swift;
...
+  }
+
+//  assert(var->isSynthesized() && "Expected compiler synthesized property");
```
Comment (DougGregor) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:73]: Probably should be removed...

**Comment by ktoso** [comment]
Sorry this one is very rough as I'm not sure if it fixes the issue and wanted to repro using a toolchain

File: lib/Sema/CodeSynthesisDistributedActor.cpp (hunk 2 above)
Comment (ktoso) [reply lib/Sema/CodeSynthesisDistributedActor.cpp:50]: Ah right, going to fix it.

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please build toolchain

**Comment by ktoso** [comment]
Not doing this anymore I think - we reworked those properties a lot

**Pull request closed without merging**